File: ocr-rn-library/ios/CountryCodeMapper.swift
//
//  CountryCodeMapper.swift
//  OCRLibrary
//
//  Shared country code mapping for OCR SDK
//

import Foundation
import UdentifyCommons

class CountryCodeMapper {
  
  static func toCountry(_ countryCode: String) -> UdentifyCommons.Country {
    switch countryCode.uppercased() {
    case "TUR", "TR", "TURKEY":
      return .TUR
    case "GBR", "GB", "UK", "UNITED_KINGDOM":
      return .GBR
    case "COL", "CO", "COLOMBIA":
      return .COL
    case "ESP", "ES", "SPAIN":
      return .ESP
    case "BRA", "BR", "BRAZIL":
      return .BRA
    case "USA", "US", "UNITED_STATES":
      return .USA
    case "PER", "PE", "PERU":
      return .PER
    case "ECU", "EC", "ECUADOR":
      return .ECU
    case "NLD", "NL", "NETHERLANDS":
      return .NLD
    case "DEU", "DE", "GERMANY":
      return .DEU
    case "FRA", "FR", "FRANCE":
      return .FRA
    case "ITA", "IT", "ITALY":
      return .ITA
    default:
      print("CountryCodeMapper - Unknown country code: \(countryCode), defaulting to TUR")
      return .TUR
    }
  }
}

File: ocr-rn-library/android/src/main/java/com/ocrrnlibrary/CountryCodeMapper.kt
package com.ocrrnlibrary

/**
 * Shared country code mapping for OCR SDK
 * Maps country codes to SDK-compatible format
 */
object CountryCodeMapper {
    
    /**
     * Converts a country code string to SDK-compatible country code
     * @param countryCode The input country code (e.g., "TUR", "TR", "TURKEY")
     * @return The SDK-compatible country code (e.g., "TUR")
     */
    fun toCountryCode(countryCode: String): String {
        return when (countryCode.uppercase()) {
            "TUR", "TR", "TURKEY" -> "TUR"
            "GBR", "GB", "UK", "UNITED_KINGDOM" -> "GBR"
            "COL", "CO", "COLOMBIA" -> "COL"
            "ESP", "ES", "SPAIN" -> "ESP"
            "BRA", "BR", "BRAZIL" -> "BRA"
            "USA", "US", "UNITED_STATES" -> "USA"
            "PER", "PE", "PERU" -> "PER"
            "ECU", "EC", "ECUADOR" -> "ECU"
            "NLD", "NL", "NETHERLANDS" -> "NLD"
            "DEU", "DE", "GERMANY" -> "DEU"
            "FRA", "FR", "FRANCE" -> "FRA"
            "ITA", "IT", "ITALY" -> "ITA"
            else -> {
                android.util.Log.w("CountryCodeMapper", "Unknown country code: $countryCode, defaulting to TUR")
                "TUR"
            }
        }
    }
}
